src/Transit/Internal/Network.hs

```diff
@@ -98,8 +98,11 @@ tcpListener :: Bool -> IO (Maybe Socket)
 tcpListener True = return Nothing
 tcpListener False = do
   let hints' = defaultHints { addrFlags = [AI_NUMERICHOST], addrSocketType = Stream }
-  addr:_ <- getAddrInfo (Just hints') (Just "0.0.0.0") (Just (show defaultPort))
-  sock' <- socket (addrFamily addr) (addrSocketType addr) (addrProtocol addr)
+  -- getAddrInfo gives us a list of addresses. With AI_ADDRCONFIG, we
```

Comment:
It looks like we didn't use AI_ADDRCONFIG though - only AI_NUMERICHOST (which is now perhaps obsolete since we don't pass an address in at all)?

Also the upstream docs sound a little different from this:

>        If hints.ai_flags includes the AI_ADDRCONFIG flag, then IPv4
>        addresses are returned in the list pointed to by res only if the
>        local system has at least one IPv4 address configured, and IPv6
>        addresses are returned only if the local system has at least one
>        IPv6 address configured.

https://man7.org/linux/man-pages/man3/getaddrinfo.3.html